core/src/main/java/org/apache/calcite/sql/SqlDialect.java

```diff
@@ -158,6 +158,7 @@
   private final Casing unquotedCasing;
   private final Casing quotedCasing;
   private final boolean caseSensitive;
+  private final boolean allowExplicitRowValueConstructor;
 
```

Comment:
Remove this `allowExplicitRowValueConstructor` flag, we expect to get this info from the `SqlConformance`.